src/runtime_src/core/common/info_aie.cpp

```diff
@@ -0,0 +1,692 @@
+/**
+ * Copyright (C) 2021 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+#define XRT_CORE_COMMON_SOURCE
+#include "info_aie.h"
+#include "core/common/query_requests.h"
+#include "core/common/device.h"
+#include <boost/algorithm/string.hpp>
+#include <boost/optional/optional.hpp>
+#include <boost/property_tree/json_parser.hpp>
+
+namespace qr = xrt_core::query;
+using ptree_type = boost::property_tree::ptree;
+
+const uint32_t major = 1;
+const uint32_t minor = 0;
+const uint32_t patch = 0;
+
+enum graph_state {
+  STOP = 0,
+  RESET = 1,
+  RUNNING = 2,
+  SUSPEND = 3,
+  END = 4
+};
+
+namespace {
+
+inline std::string
+graph_status_to_string(int status) {
+  switch(status)
+  {
+    case STOP:    return std::string("stop");
+    case RESET:   return std::string("reset");
+    case RUNNING: return std::string("running");
+    case SUSPEND: return std::string("suspend");
+    case END:     return std::string("end");
+    default:      return std::string("idle");
+  }
+}
+
+static void
+addnodelist(std::string search_str, std::string node_str,boost::property_tree::ptree& input_pt, boost::property_tree::ptree& output_pt)
+{
+  boost::property_tree::ptree pt_array;
+  for (auto& node: input_pt.get_child(search_str)) {
+    boost::property_tree::ptree pt;
+    std::string val;
+    for (auto& value: node.second)
+      val +=(val.empty()?"":", ")+ value.second.data();
+
+    pt.put("name", node.first);
+    pt.put("value", val);
+    pt_array.push_back(std::make_pair("", pt));
+  }
+  output_pt.add_child(node_str, pt_array);
+}
+
+/*
+{
+    "aie_shim": {
+        "0_0": {
+            "col": "0",
```

Comment:
Remove dead code

Reply:
This is not a dead code. This is a comments which shows how the final JSON looks like